pkl-cli/pkl-cli.gradle.kts

```diff
@@ -88,9 +187,41 @@ tasks.shadowJar {
   exclude("module-info.*")
 }
 
+val inflateProfiles by tasks.registering(Copy::class) {
+  from(zipTree(profilesZip))
+  into(layout.buildDirectory.dir("profiles"))
+  outputs.files(layout.buildDirectory.files("profiles/$profileName"))
+}
+
+val releasePrep by tasks.registering {
+  onlyIf { enableRelease }
+  dependsOn(inflateProfiles)
+  outputs.cacheIf { true }
+}
+
+fun selectNativeHostExecutable(): TaskProvider<Exec> {
+  return when {
+    buildInfo.os.isMacOsX && (buildInfo.arch == "amd64") -> macExecutableAmd64
+    buildInfo.os.isMacOsX && (buildInfo.arch == "aarch64") -> macExecutableAarch64
+    buildInfo.os.isLinux && (buildInfo.arch == "amd64") -> linuxExecutableAmd64
+    buildInfo.os.isLinux && (buildInfo.arch == "aarch64") -> linuxExecutableAarch64
+    buildInfo.os.isWindows -> windowsAmd64
+    else -> error("No host binary could be selected; please check your OS for Pkl support")
+  }
+}
+
+val nativeHostExecutable by tasks.registering {
+  selectNativeHostExecutable().get().let {
+    dependsOn(it)
+    inputs.files(it.outputs.files)
+    outputs.files(it.outputs.files)
+    outputs.cacheIf { true }
+  }
+}
+
```

Comment:
`pkl-cli` now provides a `nativeHostExecutable` task, which will use the best available native CLI executable target. This is mounted as an artifact so that smoke testing can be added for native CLIs.

Build avoidance was preserved here; running `./gradlew build` should not trigger native builds. These targets must specifically be requested with `./gradlew :pkl-cli:nativeHostExecutable` or a directly named native binary target (e.g. `macExecutableAmd64`).